### PR TITLE
LEXI-153 Create ClayAlert

### PR DESCRIPTION
**Pull request by pavel-savinov**

### CODE REVIEW

**Comment by carloslancha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: packages/clay-alert/src/ClayAlert.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	notification: Config.bool().value(false),
+
+	/**
+	 * The path to the SVG spritemap file containing the icons.
+	 * @instance
+	 * @memberof ClayIcon
+	 * @type {?string}
+	 * @default ''
+	 */
+	spritemap: Config.string().value('icons.svg'),
```
Comment (carloslancha) [inline packages/clay-alert/src/ClayAlert.js]: Spritemap never has a default value.

File: packages/clay-alert/src/ClayAlert.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	message: Config.string().required(),
+
+	/**
+	 * Show alert as notification
+	 * @instance
+	 * @memberof ClayAlert
+	 * @type {?bool}
+	 * @default false
+	 */
+	notification: Config.bool().value(false),
```
Comment (carloslancha) [inline packages/clay-alert/src/ClayAlert.js]: There are three types of alerts, Toast, Embedded and Stripe. Change this param to 
`type: Config.oneOf(['embedded', 'stripe', 'toast']).value('embedded')`

File: packages/clay-alert/src/ClayAlert.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import templates from './ClayAlert.soy.js';
+
+/**
+ * Metal Clay Alert component.
+ */
+class ClayAlert extends Component {
+	/**
+	 * @inheritDoc
+	 */
+	rendered() {
```
Comment (carloslancha) [inline packages/clay-alert/src/ClayAlert.js]: We can't do this. Each user of the component is responsible to pass a correct message following lexicon patterns, including adding correct classes to link if necessary.

File: packages/clay-alert/src/ClayAlert.soy [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	{@param message: html}
+	{@param title: string}
+	{@param? _handleCloseClick: any}
+	{@param? dismissible: bool}
+	{@param? icon: string}
+	{@param? notification: bool}
+	{@param? spritemap: string}
+	{@param? type: string}
+
+	{let $alertAttributes kind="attributes"}
```
Comment (carloslancha) [inline packages/clay-alert/src/ClayAlert.soy]: SF. https://hosting-liferayfrontendguidelines.wedeploy.io/soy/conventions.html

File: packages/clay-alert/src/ClayAlert.soy [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+		class="alert alert-{$type}{$dismissible ? ' alert-dismissible' : ''}{$notification ? ' alert-notification': ''}"
+		ref="alert"
+		role="alert"
+	{/let}
+
+	<div {$alertAttributes}>
+		{if $dismissible and $spritemap}
+            {call ClayButton.render}
```
Comment (carloslancha) [inline packages/clay-alert/src/ClayAlert.soy]: Mixed tabs & spaces.

File: packages/clay-alert/src/ClayAlert.soy [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+            	{param ariaLabel: 'Close' /}
+            	{param elementClasses: 'close' /}
+            	{param events: ['click': $_handleCloseClick] /}
+				{param icon: ['alignment': 'right', 'monospaced': false, 'spritemap': $spritemap, 'symbol': 'times'] /}
```
Comment (carloslancha) [inline packages/clay-alert/src/ClayAlert.soy]: Each param in a separate line.

File: packages/clay-alert/src/__tests__/ClayAlert.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+let alert;
+
+describe('ClayAlert', function() {
+	afterEach(() => {
+		if (alert) {
+			alert.dispose();
+		}
+	});
+
+	it('should render a alert type "info"', function() {
```
Comment (carloslancha) [inline packages/clay-alert/src/__tests__/ClayAlert.js]: We need test for each possible alert

File: packages/clay-alert/package.json [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  "esnext:main": "src/ClayAlert.js",
+  "jsnext:main": "src/ClayAlert.js",
+  "files": [
+    "lib",
+    "src",
+    "test"
+  ],
+  "scripts": {
+    "build": "npm run soy && webpack",
+    "soy": "metalsoy --soyDeps node_modules/metal*/src/**/*.soy node_modules/clay*/src/**/*.soy node_modules/clay*/node_modules/clay*/src/**/*.soy"
```
[the remark this answers is not in the document]
Comment (pavel-savinov) [reply packages/clay-alert/package.json]: @carloslancha it doesn't work this way.
Changed to this:
node_modules/clay-button/src/**/*.soy node_modules/clay-icon/src/**/*.soy

File: packages/clay-alert/webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+	},
+	output: {
+		library: 'metal',
+		libraryTarget: 'this',
+		filename: './build/globals/clay-alert.js',
+	},
+	plugins: [new webpack.optimize.ModuleConcatenationPlugin()],
+	resolve: {
+		mainFields: ['esnext:main', 'main'],
+		alias: {
```
[the remark this answers is not in the document]
Comment (pavel-savinov) [reply packages/clay-alert/webpack.config.js]: It gives JS error without alias

**Pull request closed without merging**